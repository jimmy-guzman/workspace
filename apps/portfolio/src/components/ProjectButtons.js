import React from 'react'
import PropTypes from 'prop-types'

import NextSVG from './SVGs/NextSVG'
import PrevSVG from './SVGs/PrevSVG'

const ProjectButtons = ({ renderNext, renderPrev }) => (
  <div className='projects__buttons'>
    <button
      type='button'
      className='btn__secondary'
      onClick={renderPrev}
      aria-label='Previous'
    >
      <PrevSVG height='48' width='48' />
    </button>
    <button
      type='button'
      className='btn__secondary'
      onClick={renderNext}
      aria-label='Next'
    >
      <NextSVG height='48' width='48' />
    </button>
  </div>
)

ProjectButtons.propTypes = {
  renderPrev: PropTypes.func.isRequired,
  renderNext: PropTypes.func.isRequired
}

export default ProjectButtons
