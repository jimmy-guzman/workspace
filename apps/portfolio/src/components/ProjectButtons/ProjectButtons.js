import React from 'react'
import PropTypes from 'prop-types'
import PrevSVG from '../SVGs/PrevSVG'
import NextSVG from '../SVGs/NextSVG'

export const ProjectButtons = ({ renderNext, renderPrev }) => (
  <div className='projects__buttons'>
    <button
      type='button'
      className='btn__secondary'
      onClick={renderPrev}
      aria-label='Previous'
    >
      <PrevSVG height='48' width='48' />
    </button>
    <button
      type='button'
      className='btn__secondary'
      onClick={renderNext}
      aria-label='Next'
    >
      <NextSVG height='48' width='48' />
    </button>
  </div>
)

ProjectButtons.propTypes = {
  renderPrev: PropTypes.func.isRequired,
  renderNext: PropTypes.func.isRequired
}
