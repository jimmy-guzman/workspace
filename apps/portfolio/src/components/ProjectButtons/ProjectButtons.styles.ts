import styled from 'styled-components'

import { flex, mq } from '@portfolio/utils'

export const StyledProjectButtons = styled.div`
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  ${flex('flex', 'row', 'space-between', 'center')};
  @media (min-width: ${mq('medium')}) {
    bottom: 50%;
  }
  @media (min-width: ${mq('large')}) {
    bottom: 50%;
  }
`
