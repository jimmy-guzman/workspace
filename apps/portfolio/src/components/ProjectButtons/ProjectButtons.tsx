import React from 'react'

import { PrevIcon, NextIcon } from '@portfolio/SVGs'
import { Styled } from '../Button'
import { StyledProjectButtons } from './ProjectButtons.styles'

export interface ProjectButtonsProps {
  renderPrev: () => void
  renderNext: () => void
}

export const ProjectButtons = ({
  renderNext,
  renderPrev,
}: ProjectButtonsProps) => (
  <StyledProjectButtons>
    <Styled.Button
      type='button'
      primary
      onClick={renderPrev}
      aria-label='Previous'
    >
      <PrevIcon height='48' width='48' />
    </Styled.Button>
    <Styled.Button type='button' primary onClick={renderNext} aria-label='Next'>
      <NextIcon height='48' width='48' />
    </Styled.Button>
  </StyledProjectButtons>
)
