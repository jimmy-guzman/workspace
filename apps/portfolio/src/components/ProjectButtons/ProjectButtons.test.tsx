import React from 'react'
import { render } from '@testing-library/react'

import { ProjectButtons, ProjectButtonsProps } from './ProjectButtons'

const setupProjectButtons = (
  props: ProjectButtonsProps = {
    renderPrev: jest.fn(),
    renderNext: jest.fn(),
  }
) => {
  return render(<ProjectButtons {...props} />)
}

describe('<ProjectButtons />', () => {
  it('should render default snapshot', () => {
    const { container } = setupProjectButtons()

    expect(container.firstChild).toMatchSnapshot()
  })
})
