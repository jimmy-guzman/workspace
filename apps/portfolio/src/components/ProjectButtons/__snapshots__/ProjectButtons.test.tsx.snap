// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProjectButtons /> should render default snapshot 1`] = `
.c1 {
  background: #263238;
  border: none;
  outline: none;
  font-size: 1.33em;
  font-family: 'Oswald',Impact,'Franklin Gothic Bold',sans-serif;
  cursor: pointer;
  transition: all 0.3s ease-in-out;
  color: #80CBC4;
  fill: #80CBC4;
}

.c1:hover,
.c1:active,
.c1:focus {
  color: #FFCC00;
  fill: #FFCC00;
}

.c0 {
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  display: flex;
  flex-flow: row;
  justify-content: space-between;
  align-items: center;
}

@media (min-width:769px) {
  .c0 {
    bottom: 50%;
  }
}

@media (min-width:1024px) {
  .c0 {
    bottom: 50%;
  }
}

<div
  class="c0"
>
  <button
    aria-label="Previous"
    class="c1"
    type="button"
  >
    <svg
      height="48"
      viewBox="0 0 448 512"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
        fill="currentColor"
      />
    </svg>
  </button>
  <button
    aria-label="Next"
    class="c1"
    type="button"
  >
    <svg
      height="48"
      viewBox="0 0 448 512"
      width="48"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
        fill="currentColor"
      />
    </svg>
  </button>
</div>
`;
